frontend/src/components/crd/CustomResourceList.tsx

```diff
@@ -142,20 +142,20 @@ export function CustomResourceListTable(props: CustomResourceTableProps) {
   }, [crd, apiGroup]);
 
   const cols = React.useMemo(() => {
-    const colsToDisplay = [
+    const colsToDisplay: ResourceTableProps<KubeObject<KubeCRD>>['columns'] = [
       {
         label: t('translation|Name'),
         getValue: resource => resource.metadata.name,
         render: resource => <CustomResourceLink resource={resource} crd={crd} />,
       },
-      ...additionalPrinterCols,
-      'age',
-    ] as ResourceTableProps<KubeObject<KubeCRD>>['columns'];
+    ];
 
     if (crd.isNamespacedScope) {
       colsToDisplay.splice(1, 0, 'namespace');
     }
 
+    colsToDisplay.push('cluster', ...additionalPrinterCols, 'age');
```

Comment:
Wouldn't it be more readable if we just added the `cluster` column to the original assignment above before `...additionalPrinterCols`?
Also, the cluster column is supposed to be shown only when in a multi-cluster view and I think your patch is making it show it even when there's only one cluster.